### PR TITLE
TODO and questions for the PEF

### PR DESCRIPTION
**Pull request by jjanvier**

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Model/ProductInterface.php [hunk 1]
```diff
@@ -340,36 +340,4 @@ public function getFamilyId();
      * @return ProductInterface
      */
     public function setFamilyId($familyId);
-
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Model/ProductInterface.php]: already implemented by TimestampableInterface

File: src/Pim/Bundle/CatalogBundle/Filter/AbstractFilter.php [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?php
-
+// TOOD: this whole namespace should be moved IMO to another component/bundle (AkeneoSecurity ?)
```
Comment (sumbobyboys) [inline src/Pim/Bundle/CatalogBundle/Filter/AbstractFilter.php]: Agree but not security for sure. Any other idea ?

File: src/Pim/Bundle/CatalogBundle/Filter/CollectionFilterInterface.php [hunk 3]
```diff
@@ -8,6 +9,7 @@
  * @author    Julien Sanchez <julien@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
```
Comment (sumbobyboys) [inline src/Pim/Bundle/CatalogBundle/Filter/CollectionFilterInterface.php]: fixed

File: src/Pim/Bundle/CatalogBundle/Filter/ProductValueChannelFilter.php [hunk 4]
```diff
@@ -22,6 +22,7 @@ public function filterObject($productValue, $type, array $options = [])
             throw new \LogicException('This filter only handles objects of type "ProductValueInterface"');
         }
 
+        //TODO: please rename the option and the variable to "channelCodes"
```
Comment (sumbobyboys) [inline src/Pim/Bundle/CatalogBundle/Filter/ProductValueChannelFilter.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Controller/AssociationTypeRestController.php [hunk 5]
```diff
@@ -12,6 +12,10 @@
  * @author    Filips Alpe <filips@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * TODO: maybe we could put all the product related controllers into a dedicated folder Product/
+ * TODO: do you should think controllers are usable as external REST API ? if so, maybe they should go
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Controller/AssociationTypeRestController.php]: This is clearly an internal API for now

File: src/Pim/Bundle/EnrichBundle/Controller/AssociationTypeRestController.php [hunk 6]
```diff
@@ -12,6 +12,10 @@
  * @author    Filips Alpe <filips@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * TODO: maybe we could put all the product related controllers into a dedicated folder Product/
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Controller/AssociationTypeRestController.php]: Not possible for annotations I think :/

File: src/Pim/Bundle/EnrichBundle/Controller/AttributeGroupRestController.php [hunk 7]
```diff
@@ -23,6 +23,7 @@ class AttributeGroupRestController
     protected $normalizer;
 
     /** @var CollectionFilterInterface */
+    //TODO: rename it? securityFilter? should be done in all rest controllers if we do it
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Controller/AttributeGroupRestController.php]: yeap, sure

File: src/Pim/Bundle/EnrichBundle/Controller/AttributeRestController.php [hunk 8]
```diff
@@ -47,6 +47,7 @@ public function __construct(
      */
     public function indexAction()
     {
+        //TODO: is it normal? no performances pbs?
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Controller/AttributeRestController.php]: tested with 2000 attributes and the grid is way behind the pef on this subject. Anyway, on ref environments , we are around 3 seconds to load the first pef and no overhead after.

File: src/Pim/Bundle/CatalogBundle/Resolver/FQCNResolver.php [hunk 9]
```diff
@@ -8,6 +8,9 @@
  * @author    Julien Sanchez <julien@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * //TODO: I don't like this object at all
+ * //TODO: and even if we really need it, that should not be here
```
Comment (sumbobyboys) [inline src/Pim/Bundle/CatalogBundle/Resolver/FQCNResolver.php]: fixed with the container. Any suggestions for the new location ?

File: src/Pim/Bundle/CatalogBundle/Resources/config/filters.yml [hunk 10]
```diff
@@ -4,12 +4,14 @@ parameters:
     pim_enrich.filter.product_value.channel.class: Pim\Bundle\CatalogBundle\Filter\ProductValueChannelFilter
 
 services:
+    # TODO: should be moved to a dedicated bundle (AkeneoSecurity ?)
```
Comment (sumbobyboys) [inline src/Pim/Bundle/CatalogBundle/Resources/config/filters.yml]: agree but where ?

File: src/Pim/Bundle/CatalogBundle/Resources/config/filters.yml [hunk 11]
```diff
@@ -4,12 +4,14 @@ parameters:
...
     pim_catalog.filter.chained:
         class: %pim_catalog.filter.chained.class%
 
     pim_enrich.filter.product_value.locale:
         class: %pim_enrich.filter.product_value.locale.class%
         tags:
+            #TODO: we never use double dots on tags, do we ? maybe we should replace them by "." to be consistent
```
Comment (sumbobyboys) [inline src/Pim/Bundle/CatalogBundle/Resources/config/filters.yml]: fixed

File: src/Pim/Bundle/EnrichBundle/Controller/FamilyRestController.php [hunk 12]
```diff
@@ -38,6 +38,7 @@ public function __construct(EntityRepository $familyRepository, NormalizerInterf
      */
     public function indexAction()
     {
+        //TODO: is it normal? no performances pbs?
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Controller/FamilyRestController.php]: as it works with 2000 attributes and there is never as many families it should be fine. This is still an unoptimised version and we can fix that later. For now seems like a premature optimisation

File: src/Pim/Bundle/EnrichBundle/Controller/GroupRestController.php [hunk 13]
```diff
@@ -41,6 +41,7 @@ public function __construct(GroupManager $groupManager, NormalizerInterface $nor
      */
     public function indexAction()
     {
+        //TODO: inject the repo instead
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Controller/GroupRestController.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Controller/GroupRestController.php [hunk 14]
```diff
@@ -53,6 +54,8 @@ public function indexAction()
      */
     public function getAction($identifier)
     {
+        //TODO: inject the repo instead
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Controller/GroupRestController.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Controller/GroupRestController.php [hunk 15]
```diff
@@ -53,6 +54,8 @@ public function indexAction()
...
+        //TODO: findOneBy(['code' => ...])
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Controller/GroupRestController.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Controller/GroupRestController.php [hunk 16]
```diff
@@ -69,6 +72,8 @@ public function getAction($identifier)
      */
     public function listProductsAction($identifier)
     {
+        //TODO: inject the repo instead
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Controller/GroupRestController.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Controller/GroupRestController.php [hunk 17]
```diff
@@ -69,6 +72,8 @@ public function getAction($identifier)
...
+        //TODO: findOneBy(['code' => ...])
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Controller/GroupRestController.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Controller/ProductCommentRestController.php [hunk 18]
```diff
@@ -19,6 +19,9 @@
  * @author    Filips Alpe <filips@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * TODO: a lot of things of CommentController have been removed and should not have been removed
+ * TODO: now we can not use comments out of products
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Controller/ProductCommentRestController.php]: yes you can. You can use the pef extension anywhere without any problem.

File: src/Pim/Bundle/EnrichBundle/Normalizer/CollectionNormalizer.php [hunk 19]
```diff
@@ -12,15 +12,20 @@
  * @author    Julien Sanchez <julien@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * TODO: spec it !!
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/CollectionNormalizer.php]: done

File: src/Pim/Bundle/EnrichBundle/Normalizer/CollectionNormalizer.php [hunk 20]
```diff
@@ -12,15 +12,20 @@
...
  */
 class CollectionNormalizer implements NormalizerInterface, SerializerAwareInterface
 {
-    /** @var SerializerInterface $serializer */
+    /** @var SerializerInterface */
+    //TODO: we need a normalizer here
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/CollectionNormalizer.php]: not possible

File: src/Pim/Bundle/EnrichBundle/Normalizer/CollectionNormalizer.php [hunk 21]
```diff
@@ -12,15 +12,20 @@
...
     protected $serializer;
 
     /** @var array */
     protected $supportedFormat = ['internal_api'];
 
+    //TODO: no constructor?
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/CollectionNormalizer.php]: nope

File: src/Pim/Bundle/EnrichBundle/Normalizer/CollectionNormalizer.php [hunk 22]
```diff
@@ -40,6 +45,7 @@ public function normalize($elements, $format = null, array $context = array())
      */
     public function supportsNormalization($data, $format = null)
     {
+        //TODO \Iterable does not exist ^^
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/CollectionNormalizer.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Normalizer/CompletenessCollectionNormalizer.php [hunk 23]
```diff
@@ -10,6 +10,8 @@
  * @author    Julien Sanchez <julien@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * TODO: spec it !!
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/CompletenessCollectionNormalizer.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Normalizer/CompletenessNormalizer.php [hunk 24]
```diff
@@ -11,6 +11,8 @@
  * @author    Julien Sanchez <julien@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * TODO: spec it !!
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/CompletenessNormalizer.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Normalizer/GroupNormalizer.php [hunk 25]
```diff
@@ -11,6 +11,8 @@
  * @author    Filips Alpe <filips@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * TODO: spec it
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/GroupNormalizer.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Normalizer/LocaleNormalizer.php [hunk 26]
```diff
@@ -12,6 +12,8 @@
  * @author    Julien Sanchez <julien@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * TODO: spec it !!
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/LocaleNormalizer.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Normalizer/ProductNormalizer.php [hunk 27]
```diff
@@ -15,6 +15,8 @@
  * @author    Julien Sanchez <julien@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * TODO: spec it
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/ProductNormalizer.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Normalizer/ProductNormalizer.php [hunk 28]
```diff
@@ -28,6 +30,7 @@ class ProductNormalizer implements NormalizerInterface, SerializerAwareInterface
     protected $versionManager;
 
     /** @var SerializerInterface */
+    //TODO: we need a normalizer here
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/ProductNormalizer.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Normalizer/ProductNormalizer.php [hunk 29]
```diff
@@ -36,6 +39,7 @@ class ProductNormalizer implements NormalizerInterface, SerializerAwareInterface
      * @param LocaleManager       $localeManager
      */
     public function __construct(
+        //TODO: we need a normalizer here
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/ProductNormalizer.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Normalizer/SequentialEditNormalizer.php [hunk 30]
```diff
@@ -11,6 +11,8 @@
  * @author    Filips Alpe <filips@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * TODO: spec it
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/SequentialEditNormalizer.php]: fixed

File: src/Pim/Bundle/EnrichBundle/Normalizer/VersionNormalizer.php [hunk 31]
```diff
@@ -13,6 +13,8 @@
  * @author    Julien Sanchez <julien@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * TODO: spec it
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/VersionNormalizer.php]: fixed

**Pull request closed without merging**